src/library/scala/sys/process/ProcessBuilder.scala

```diff
@@ -173,6 +173,17 @@ trait ProcessBuilder extends Source with Sink {
     */
   def lineStream: Stream[String]
 
+  /** Starts the process represented by this builder.  The output is returned as
+   * a Stream that blocks when lines are not available but the process has not
+   * completed.
+   * The producer process will block if the given capacity of lines if filled
+   * without being consumed from the stream.
```

Comment:
Can you provide more documentation what happens in this case? Is the child process suspended until the stream is being consumed? Or is the output buffered somewhere else? Or a mix of these?

Reply:
The thread in scala that consumes from the child will get blocked on inserting to LinkedBlockingQueue, until the application consumes the stream.

The behaviour of the child process is OS dependent, but similar anywhere.
On a POSIX system the child and parent process communicate through a PIPE, which on Linux has a 64kb buffer by default. If that buffer is full, the child process will be put to sleep by the kernel and made runnable again after the parent process consumption makes space in the buffer.
This is the standard behaviour of interprocess communication in any OS, so in my opinion it seems out of scope here.

Reply:
OK, thanks for the background!